src/components/editor/Store.tsx

```diff
@@ -54,7 +55,14 @@ const getInitialState = <T,>(
         setSpecs: (newVal) => {
             set(
                 produce((state) => {
-                    state.specs = newVal;
+                    if (newVal && newVal.length > 0) {
+                        if (state.specs === null) {
+                            console.log('updating current catalog');
+
+                            state.currentCatalog = newVal[0];
+                        }
+                        state.specs = newVal;
+                    }
```

Comment:
When specs are set the first time we want to set the current catalog to the first one in the list.